velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -276,7 +427,8 @@ std::unique_ptr<ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
           maxDefine,
           curSchemaIdx,
           ++schemaIdx,
-          columnIdx);
+          columnIdx,
+          fileMetaData_->schema);
```

Comment:
@agrawalreetika I see you add the recursive name updating code updateColumnNames() instead of updating them in place in getParquetColumnInfo(). Have you tried that option? It is supposed to reduce the amount of code because we don't need to traverse the tree again. Instead we can build the tree with updated column names on the fly, while constrcuting the schema tree. Was it indeed too much change? Since you took the first option, I don't see any need for this new parameter schema.